crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -3357,4 +3336,960 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn transaction_id_found_in_database() -> eyre::Result<()> {
+        let mut rng = generators::rng();
+        let (provider, database_blocks, _, _) = provider_with_random_blocks(
+            &mut rng,
+            TEST_BLOCKS_COUNT,
+            0,
+            BlockRangeParams {
+                tx_count: TEST_TRANSACTIONS_COUNT..TEST_TRANSACTIONS_COUNT,
+                ..Default::default()
+            },
+        )?;
```

Comment:
this setup can insert data in both database and memory, so let's combine the `transaction_id` tests for both database and in-memory into one `transaction_id` test? same as it's done for some of the tests above where we assert both database and in-memory in one test.

Reply:
I just fixed it. I didn't do this at first to try to isolate the tests and avoid too many asserts per test. But with the large number of tests, I gathered the memory/database/not found into a single test each time. That's quite a few asserts each time but it should work, wdyt?

Reply:
I think too many asserts per test isn't an issue, because you will see what assert the test fails in the logs anyway. IMO having one test function per provider method is fine.

Reply:
It should be good now. I think there is only one method that I cut into two tests otherwise it was way too long. But overall I shortened everything